ambry-clustermap/src/main/java/com.github.ambry.clustermap/PartitionLayout.java

```diff
@@ -220,15 +219,16 @@ private long getNewPartitionId() {
   }
 
   // Creates a Partition and corresponding Replicas for each specified disk
-  public Partition addNewPartition(List<Disk> disks, long replicaCapacityInBytes, String partitionClass) {
+  public Partition addNewPartition(List<Disk> disks, long replicaCapacityInBytes, String partitionClass,
```

Comment:
the `ClusterMapConfig` need not be passed in. You can get it in this class using `hardwareLayout.getClusterMapConfig()`

Reply:
yeah, you are right. Let me check all the places and remove unnecessary `ClusterMapConfig`.